rtil/src/threads/ue/rebo/livesplit.rs

```diff
@@ -97,4 +110,34 @@ impl Run {
     pub fn attempt_count() -> u32 {
         LIVESPLIT_STATE.lock().unwrap().timer.run().clone().attempt_count()
     }
+    pub fn save_splits(path: impl AsRef<Path>) -> Result<(), SplitsSaveError> {
+        let state = LIVESPLIT_STATE.lock().unwrap();
+        let path = path.as_ref();
+        let filename = path.file_name().expect("Could not get filename").to_str().expect("Could not convert filename to &str");
+        let file = match File::create(path) {
+            Ok(file) => file,
+            Err(e) => return Err(SplitsSaveError::CreationError(filename.to_owned(), e.to_string())),
+        };
+        let writer = BufWriter::new(file);
+        match save_timer(&state.timer, IoWrite(writer)) {
+            Ok(_) => Ok(()),
+            Err(e) => Err(SplitsSaveError::SaveError(filename.to_owned(), e.to_string())),
+        }
```

Comment:
This is a bit of a matter of taste, but I like it more if this match has the same structure of the previous match with an early return in the error-case and then move out the implicit `Ok(())` return.